packages/components/src/public/QueryModel/withQueryModels.tsx

```diff
@@ -1145,6 +1211,48 @@ export function withQueryModels<Props>(
             );
         };
 
+        /**
+         * A method used to indicate that a user has made a change to the underlying data of a QueryModel, e.g. added,
+         * updated, or deleted rows. This method will at a minimum reload the model, but may also reset the pagination
+         * if the change could result in a bad state. See Issue 51987.
+         * @param id the id of the model
+         * @param modelChange a ModelChange object describing the change
+         */
+        onModelChange = (id: string, modelChange: ModelChange): void => {
```

Comment:
I really like having the consolidation of this stuff here in withQueryModel so that we don't have to remember all of the scenarios to check for with each usage!